modules/bucket-events/README.md

```diff
@@ -0,0 +1,105 @@
+# `bucket-events`
+
+This module provisions infrastructure to listen to events from a GCS bucket and
+publish them to a broker.
+
+```hcl
+// Create a network with several regional subnets
+module "networking" {
+  source = "chainguard-dev/common/infra//modules/networking"
+
+  name       = "my-networking"
+  project_id = var.project_id
+  regions    = [...]
+}
+
+// Create the Broker abstraction.
+module "cloudevent-broker" {
+  source = "chainguard-dev/common/infra//modules/cloudevent-broker"
+
+  name       = "my-broker"
+  project_id = var.project_id
+  regions    = module.networking.regional-networks
+}
+
+// Get the bucket.
+data "google_storage_bucket" "bucket" {
+  name = "my-bucket"
+}
+
+// Listen to events.
+module "bucket-events" {
+  source = "chainguard-dev/common/infra//modules/bucket-events"
+
+  name       = "my-broker"
+  project_id = var.project_id
+
+  bucket   = data.google_storage_bucket.bucket.name
```

Comment:
If you just pass this in, then the module can use the `data` source and you can simplify the interface (drop location)